### PR TITLE
Add husky

### PR DESCRIPTION
**Pull request by iHiD**
@kntsoriano @ErikSchierboom This adds pre-commit support (via husky) to run RuboCop on Ruby files and Prettier on everything else.

### CODE REVIEW

File: app/controllers/pages_controller.rb [hunk 1]
```diff
@@ -1,6 +1,7 @@
 class PagesController < ApplicationController
   def index
     solution = Solution.first
+
     IterationsChannel.broadcast!(solution)
   end
-end
+end
```
Comment (kotp) [inline app/controllers/pages_controller.rb:7]: Trailing white space introduced here.

**Comment by kotp** [comment]
Well, I was in the middle of the review… too late now. :rofl: Missed it by seconds.

**Comment by iHiD** [comment]
@kotp Sorry :) Feel free to leave comments :)

File: app/controllers/pages_controller.rb (hunk 1 above)
Comment (iHiD) [reply app/controllers/pages_controller.rb:7]: Hmmm. I don't understand why this didn't get picked up by CI. More debugging needed!

File: app/controllers/pages_controller.rb [hunk 2]
```diff
@@ -1,6 +1,7 @@
 class PagesController < ApplicationController
   def index
     solution = Solution.first
+
     IterationsChannel.broadcast!(solution)
```
Comment (kotp) [inline app/controllers/pages_controller.rb:5]: Unless `solution` is used more than once, this is a documenting variable.

`IterationsChannel.broadcast!(Solution.first)` is very clear without it.

**Pull request merged**